### PR TITLE
[3.0] Judge whether success or not by user's perspective in interface-appName mapping.

### LINKED ISSUE #8722: [3.0] the same service export multi protocol error
- [ ] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [ ] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 3.0.3-SNAPSHOT
* Operating System version: macos
* Java version: 1.8

### Steps to reproduce this issue

1.  provider export multi protocol 
```java
class ApiMigrationBothProvider {

    public static void main(String[] args) {

        ServiceConfig<IWrapperGreeter> service = new ServiceConfig<>();
        service.setInterface(IWrapperGreeter.class);
        service.setRef(new IGreeter2Impl());

        DubboBootstrap bootstrap = DubboBootstrap.getInstance();
        bootstrap.application(new ApplicationConfig("demo-migration-both-provider"))
                .registry(new RegistryConfig(TriSampleConstants.ZK_ADDRESS))
                .protocol(new ProtocolConfig(CommonConstants.TRIPLE, TriSampleConstants.SERVER_PORT))
                .protocol(new ProtocolConfig(CommonConstants.DUBBO, TriSampleConstants.DEFAULT_DUBBO_PORT))
                .service(service)
                .start()
                .await();

    }
}
```

2. run provider

## reason

image
 
when the second protocol export, contains always true 

Pls. provide [GitHub address] to reproduce this issue.

https://github.com/apache/dubbo-samples/pull/356

### Expected Result

everything is ok

### Actual Result

throw exception

If there is an exception, please attach the exception trace:

```
java.lang.RuntimeException
	at org.apache.dubbo.registry.client.metadata.MetadataServiceNameMapping.map(MetadataServiceNameMapping.java:88)
	at org.apache.dubbo.config.ServiceConfig.lambda$exported$1(ServiceConfig.java:270)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.apache.dubbo.config.ServiceConfig.exported(ServiceConfig.java:266)
	at org.apache.dubbo.config.ServiceConfig.doExport(ServiceConfig.java:371)
	at org.apache.dubbo.config.ServiceConfig.export(ServiceConfig.java:244)
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.exportServices(DubboBootstrap.java:1476)
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.doStart(DubboBootstrap.java:1190)
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.start(DubboBootstrap.java:1166)
	at org.apache.dubbo.sample.tri.migration.ApiMigrationBothProvider.main(ApiMigrationBothProvider.java:44)
```

### PR DESCRIPTION
**Pull request by plusmancn**
## What is the purpose of the change
* Judge whether success or not by user's perspective in interface-appName mapping.
* By the way, format the function `MetadataServiceNameMapping#map`: change the indent spaces from 8 to 4.

Fixes #8722 

## Brief changelog
Every time the provider restarted after the first start,  it would raise an error "Failed register interface application mapping......".
But from the user's perspective, it means successful when the oldConfigContent has contained the current appName. So we should not throw an Exception to the user, it will confuse the user.

Just change one line code：
```java
// => org.apache.dubbo.registry.client.metadata.MetadataServiceNameMapping#map
public boolean map(URL url) {
    // ......
    if (StringUtils.isNotEmpty(oldConfigContent)) {
        boolean contains = StringUtils.isContains(oldConfigContent, appName);
        if (contains) {
            // From user's perspective, it means successful when the oldConfigContent has contained the current appName. So we should not throw an Exception to user, it will confuse user.
            succeeded = true;   // <= The only changes for this function.
            break;
        }
        newConfigContent = oldConfigContent + COMMA_SEPARATOR + appName;
    }
    // ......
}
```

What's more, the `map` function is only called by `ServiceConfig#export`，the changes are safe.
```java
// => org.apache.dubbo.config.ServiceConfig#export
protected void exported() {
    exported = true;
    List<URL> exportedURLs = this.getExportedUrls();
    exportedURLs.forEach(url -> {
        if (url.getParameters().containsKey(SERVICE_NAME_MAPPING_KEY)) {
            ServiceNameMapping serviceNameMapping = ServiceNameMapping.getDefaultExtension(getScopeModel());
            try {
                // try to register
                boolean succeeded = serviceNameMapping.map(url);
                if (succeeded) {
                    logger.info("Successfully registered interface application mapping for service " + url.getServiceKey());
                }
            } catch (Exception e) {
                logger.error("Failed register interface application mapping for service " + url.getServiceKey(), e);
            }
        }
    });
    onExported();
}
```

### CODE REVIEW

**Review by guohao — CHANGES_REQUESTED** [review]
PLZ do not format all code ... Just modify the small part

**Comment by EarthChen** [comment]
#8722 #8721

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/MetadataServiceNameMapping.java [hunk 1]
```diff
@@ -53,38 +53,40 @@ public MetadataServiceNameMapping() {
...
+        boolean succeeded = false;
+        String newConfigContent = appName;
+        do {
+            ConfigItem configItem = metadataReport.getConfigItem(serviceInterface, DEFAULT_MAPPING_GROUP);
+            String oldConfigContent = configItem.getContent();
+            if (StringUtils.isNotEmpty(oldConfigContent)) {
+                boolean contains = StringUtils.isContains(oldConfigContent, appName);
+                if (contains) {
+                    // From the user's perspective, it means successful when the oldConfigContent has contained the current appName. So we should not throw an Exception to user, it will confuse the user.
+                    succeeded = true;
```
Comment (EarthChen) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/MetadataServiceNameMapping.java:82]: Is it better to just return true?

**Review by EarthChen — CHANGES_REQUESTED** [review]

**Comment by plusmancn** [comment]
> PLZ do not format all code ... Just modify the small part

@guohao  The original style is unreadable, see follow:

image

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/MetadataServiceNameMapping.java (hunk 1 above)
Comment (plusmancn) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/MetadataServiceNameMapping.java:82]: According to the original function,  setting the `succeeded` variable may be better.
The latter code may do something base on `succeeded` status.

Comment (EarthChen) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/MetadataServiceNameMapping.java:82]: Based on existing code, a direct return is better

**Review by guohao — APPROVED** [review]

**Pull request merged**